components/micro-gateway-core/src/main/ballerina/src/gateway/listeners/api_gateway_listener.bal

```diff
@@ -120,11 +120,19 @@ public function getAuthHandlers() returns http:InboundAuthHandler[] {
         },
         verifyHostname: getConfigBooleanValue(HTTP_CLIENTS_INSTANCE_ID, ENABLE_HOSTNAME_VERIFICATION, true)
     };
+
     http:OutboundAuthConfig? auth = ();
-    if (getConfigBooleanValue(KM_CONF_SECURITY_BASIC_INSTANCE_ID, ENABLED, DEFAULT_KM_CONF_SECURITY_BASIC_ENABLED)) {
+    // support backward compatibility in reading the basic auth credentials when connecting with KM.
+    string username = getConfigValue(KM_CONF_INSTANCE_ID, USERNAME, "");
+    string password = getConfigValue(KM_CONF_INSTANCE_ID, PASSWORD, "");
```

Comment:
```suggestion
    string password = getConfigValue(KM_CONF_INSTANCE_ID, PASSWORD, DEFAULT_PASSWORD);
```

Reply:
We can't have default values there. This is to support the old config. The empty string determines that old config is not used